### PR TITLE
[Ldap] Add verbose ext-ldap error if present for easier debugging

### PR DESCRIPTION
**Pull request by scaytrase**
| Q             | A
| ------------- | ---
| Branch?       | master (be careful when merging)
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no   
| Deprecations? | no
| Tests pass?   | 
| Fixed tickets | #28149
| License       | MIT
| Doc PR        | 

Added an optional message suffix if ldap_errno is not equals to 0

### CODE REVIEW

File: src/Symfony/Component/Ldap/Adapter/ExtLdap/Query.php [hunk 1]
```diff
@@ -89,7 +89,12 @@ public function execute()
         }
 
         if (false === $this->search) {
-            throw new LdapException(sprintf('Could not complete search with dn "%s", query "%s" and filters "%s".', $this->dn, $this->query, implode(',', $this->options['filter'])));
+            $ldapError = '';
+            if (0 !== ($errno = ldap_errno($con))) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Ldap/Adapter/ExtLdap/Query.php]: I'd suggest to write this as `if ($errno = ldap_errno($con)) {`

**Review by nicolas-grekas — APPROVED** [review]
(with minor comment)

**Comment by nicolas-grekas** [comment]
(to be merged on master actually as that's not a bug fix but an improvement.)

File: src/Symfony/Component/Ldap/Adapter/ExtLdap/Query.php (hunk 1 above)
Comment (scaytrase) [reply src/Symfony/Component/Ldap/Adapter/ExtLdap/Query.php]: I hoped to find something like `LDAP_ERROR_NONE` for better readability, but failed. Initial idea was

```
if (LDAP_ERROR_NONE !== ($errno = ldap_errno($con))) {
```

Without that your suggestion looks better, indeed, I'll fix it

**Comment by scaytrase** [comment]
@nicolas-grekas should I rebase?

**Comment by nicolas-grekas** [comment]
@scaytrase if you can please yes, otherwise no worries we'll take care of it.

**Comment by scaytrase** [comment]
rebased

**Comment by nicolas-grekas** [comment]
Thank you @scaytrase.

**Pull request merged**